jerry-main/main-unix.c

```diff
@@ -627,6 +640,21 @@ main (int argc,
 
   jerry_release_value (ret_value);
 
+  jerry_value_t gc_value = jerry_create_external_function (gc_handler);
+  jerry_value_t gc_func_name_val = jerry_create_string ((jerry_char_t *) "__gc");
+  ret_value = jerry_set_property (global_obj_val, gc_func_name_val, gc_value);
+  jerry_release_value (gc_func_name_val);
+  jerry_release_value (gc_value);
+
+  if (jerry_value_has_error_flag (ret_value))
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_WARNING, "Warning: failed to register '__gc' method.");
+    print_unhandled_exception (ret_value);
+  }
+
+  jerry_release_value (ret_value);
```

Comment:
The registration of an external function should be factored out into a helper function. The steps are always the same: create the function, create the string for its name, register it in the global object, handle failure, release everything.